From 644075762fae4d1dd34f9c4f9a9885d94f823f2c Mon Sep 17 00:00:00 2001
From: riddle_hsu <riddle_hsu@htc.com>
Date: Fri, 15 Apr 2016 12:05:06 +0800
Subject: [PATCH 100/296] Fix NPE from AppOpsManager.checkPackage.

Since commit 0fcef84, any application can use below code
to insert an invalid op record:
    appOpsManager.checkPackage(0, null);

Use below command will see the invalid record breaks writeState.
  adb shell dumpsys appops write-settings

  Exception occurred while dumping:
  java.lang.NullPointerException: Attempt to invoke virtual method
  'boolean java.lang.String.equals(java.lang.Object)'
  on a null object reference
        at com.android.server.AppOpsService.writeState

And system server will crash in 30 minutes when saving the xml.
---
 services/core/java/com/android/server/AppOpsService.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/AppOpsService.java b/services/core/java/com/android/server/AppOpsService.java
index 91509f63ff4..b3a62cb0a14 100644
--- a/services/core/java/com/android/server/AppOpsService.java
+++ b/services/core/java/com/android/server/AppOpsService.java
@@ -1128,7 +1128,7 @@ public class AppOpsService extends IAppOpsService.Stub {
     public int checkPackage(int uid, String packageName) {
         Preconditions.checkNotNull(packageName);
         synchronized (this) {
-            if (getOpsRawLocked(uid, packageName, true) != null) {
+            if (packageName != null && getOpsRawLocked(uid, packageName, true) != null) {
                 return AppOpsManager.MODE_ALLOWED;
             } else {
                 return AppOpsManager.MODE_ERRORED;
-- 
2.11.0

